src/main/java/org/generationcp/middleware/service/impl/inventory/LotServiceImpl.java

```diff
@@ -115,21 +119,66 @@ public String saveLot(final CropType cropType, final Integer userId, final LotGe
 	}
 
 	@Override
-	public void updateLots(final List<ExtendedLotDto> lotDtos, final LotUpdateRequestDto lotRequest) {
+	public void updateLots(final List<ExtendedLotDto> lotDtos, final LotUpdateRequestDto lotUpdateRequestDto) {
+		if (lotUpdateRequestDto.getSingleInput() != null) {
+			this.updateLots(lotDtos, lotUpdateRequestDto.getSingleInput());
+		} else {
+			this.updateLots(lotDtos, lotUpdateRequestDto.getMultiInput());
+		}
+	}
+
+	private void updateLots(final List<ExtendedLotDto> lotDtos, final LotSingleUpdateRequestDto lotSingleUpdateRequestDto){
 		final LotDAO lotDao = this.daoFactory.getLotDao();
 		for (final LotDto lotDto : lotDtos) {
 			final Lot lot = lotDao.getById(lotDto.getLotId());
-			if (lotRequest.getGid() != null) {
-				lot.setEntityId(lotRequest.getGid());
+			if (lotSingleUpdateRequestDto.getGid() != null) {
+				lot.setEntityId(lotSingleUpdateRequestDto.getGid());
+			}
+			if (lotSingleUpdateRequestDto.getLocationId() != null) {
+				lot.setLocationId(lotSingleUpdateRequestDto.getLocationId());
 			}
-			if (lotRequest.getLocationId() != null) {
-				lot.setLocationId(lotRequest.getLocationId());
+			if (lotSingleUpdateRequestDto.getUnitId() != null) {
+				lot.setScaleId(lotSingleUpdateRequestDto.getUnitId());
+			}
+			if (!StringUtils.isBlank(lotSingleUpdateRequestDto.getNotes())) {
+				lot.setComments(lotSingleUpdateRequestDto.getNotes());
+			}
+			lotDao.save(lot);
+		}
+	}
+
+	private void updateLots(final List<ExtendedLotDto> lotDtos, final LotMultiUpdateRequestDto lotMultiUpdateRequestDto) {
+		final LotDAO lotDao = this.daoFactory.getLotDao();
+
+		final List<Location> locations = this.daoFactory.getLocationDAO()
+			.filterLocations(STORAGE_LOCATION_TYPE, null, lotMultiUpdateRequestDto.getLotList().stream()
+				.map(LotUpdateDto::getStorageLocationAbbr)
+				.collect(Collectors.toList()));
+		// locationsMapByLocationAbbr
+		final Map<String, Integer> locationsMapByLocationAbbr =
+			locations.stream().collect(Collectors.toMap(Location::getLabbr, Location::getLocid));
+
+		final VariableFilter variableFilter = new VariableFilter();
+		variableFilter.addPropertyId(TermId.INVENTORY_AMOUNT_PROPERTY.getId());
+		final List<Variable> unitVariables = this.ontologyVariableDataManager.getWithFilter(variableFilter);
+		// unitMapByName
+		final Map<String, Integer> unitMapByName =
+			unitVariables.stream().collect(Collectors.toMap(Variable::getName, Variable::getId));
+		final Map<String, LotUpdateDto> LotUpdateMapByLotUID =
+			Maps.uniqueIndex(lotMultiUpdateRequestDto.getLotList(), LotUpdateDto::getLotUID);
+
+		for (final LotDto lotDto : lotDtos) {
+			final Lot lot = lotDao.getById(lotDto.getLotId());
```

Comment:
It would be better if we make only one call to the database to get all the lots you want to update and have a map by UUID with the lotDtos to find the new values